### PR TITLE
Allow for read of tlat, tlon, anglet with popgrid

### PR DESCRIPTION
**Pull request by TillRasmussen**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [ ] Short (1 sentence) summary of your PR: 
   Eliminate warning due to print of iblk when not in a loop 
- [ ] Developer(s): 
    Till Rasmussen, DMI
- [ ] Suggest PR reviewers from list in the column to the right.
  @apcraig 
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    ENTER INFORMATION HERE 
- How much do the PR code changes differ from the unmodified code? 
    - [x ] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ x] No 
- [ ] Please provide any additional information or relevant details below:

### CODE REVIEW

**Comment by TillRasmussen** [comment]
The blank spaces is a result of some test I have been doing in order to read anglet, tlon and tlat instead of calculating these. This included adding some if else around the Tlonlat function..

**Comment by eclare108213** [comment]
@TillRasmussen is this PR ready to merge or are you still working on it?
Is the call to ice_barrier necessary?

File: cicecore/cicedynB/infrastructure/ice_grid.F90 [hunk 1]
```diff
@@ -820,18 +830,44 @@ subroutine popgrid_nc
...
+         where (ANGLET < -pi) ANGLET = -pi
+          fieldname="tlon"
+          call ice_read_global_nc(fid_grid,1,fieldname,work_g1,diag)
+          call scatter_global(TLON, work_g1, master_task, distrb_info, &
+                             field_loc_center, field_type_scalar)
+          fieldname="tlat"
+          call ice_read_global_nc(fid_grid,1,fieldname,work_g1,diag)
+          call scatter_global(TLAT, work_g1, master_task, distrb_info, &
+                             field_loc_center, field_type_scalar)
+      endif
```
Comment (apcraig) [inline cicecore/cicedynB/infrastructure/ice_grid.F90:863]: I looks like there might be a slight indentation problem in this if block.  Can you check that there are no "tabs".

**Comment by TillRasmussen** [comment]
This pull request has been updated. Editiing is done if test are complete
1/ To fix bug in ice_Transport driver .F90 (original pull request)
2/ Fix bug when cpp flag NEMO_IN_CICE is enabled. ice_step_mod.F90 uses variable raice without declaring.3
3/ Allow for reading angle, tlon tlat if angle exist in netcdf grid file.  If these are read they will not be calculated. This will cause differences when either of these three variables are used or dumped to output. If angle do not exist the routine will calculate angle, tlon and tlat as previous.

File: cicecore/cicedynB/infrastructure/ice_grid.F90 (hunk 1 above)
Comment (TillRasmussen) [reply cicecore/cicedynB/infrastructure/ice_grid.F90:849]: I thnk that this has been solved

**Comment by TillRasmussen** [comment]
My icepack fork is now updated

**Comment by apcraig** [comment]
I ran a full test suite last night and everything looks good, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash#3fedc78b1b012f8bd6c0fde7c1b41e79b9db63a9

**Pull request merged**